llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp

```diff
@@ -428,6 +431,7 @@ bool SPIRVInstructionSelector::spvSelect(Register ResVReg,
 
   case TargetOpcode::G_INTRINSIC:
   case TargetOpcode::G_INTRINSIC_W_SIDE_EFFECTS:
+  case TargetOpcode::G_INTRINSIC_CONVERGENT:
```

Comment:
Is there anything that distinguishes a `G_INTRINSIC_CONVERGENT` vs a `G_INTRINSIC`?  Seems like we are handling them the same way so wondering what happens with that extra information about convergence?

Reply:
Afaict, not within this section of the codebase, it is used elsewhere though eg `GlobalISel`. But I think @Keenuts would be able to better elaborate.

Reply:
The extra convergence information is being used when we generate the merge instructions. They should not change how the instrinsic itself is generated. The convergence token help us distinguish between:

```
int i;
for(i = 0; i < N; ++i) {
  if (cond(i)) {
    // The "active" lanes are those where cond was true in the same iteration.
    WaveReadLaneAt(i);
    break;
  }
}


for(i = 0; i < N; ++i) {
  if (cond(i))
    break;
}
// All lanes are active because they all reconverged at the end of the loop.
WaveReadLaneAt(i);
```